packages/search-adapter/src/empathy/mappers/response/facets/empathy-hierarchical-facet.mapper.ts

```diff
@@ -30,16 +30,25 @@ export class EmpathyHierarchicalFacetMapper implements ResponseMapper<EmpathyFac
   }
 
   protected mapFiltersTree(facet: HierarchicalFacet, rawFilters: EmpathyFilter[], context: ResponseMapperContext): HierarchicalFilter[] {
-    const initFilterProps: Partial<HierarchicalFilter> = { facetId: facet.id, parentId: null };
-    return this.mapDeepFilters(rawFilters, initFilterProps, context);
+    const filters: HierarchicalFilter[] = [];
+    const initFilterProps: Readonly<Partial<HierarchicalFilter>> = { facetId: facet.id, parentId: null };
+    rawFilters.forEach(rawFilter => {
+      this.mapDeepFilter(rawFilter, initFilterProps, context, filters);
+    });
+    return filters;
   }
 
-  protected mapDeepFilters(rawFilters: EmpathyFilter[] = [], initFilterProps: Readonly<Partial<HierarchicalFilter>>,
-    context: ResponseMapperContext): HierarchicalFilter[] {
-    return rawFilters.map(rawFilter => {
-      const filter = this.mapFilter(rawFilter, { ...initFilterProps } as HierarchicalFilter, context);
-      filter.children = this.mapDeepFilters(rawFilter.values, { ...initFilterProps, parentId: filter.id }, context);
-      return filter;
-    });
+  protected mapDeepFilter(
+    rawFilter: EmpathyFilter,
+    initFilterProps: Readonly<Partial<HierarchicalFilter>>,
+    context: ResponseMapperContext,
+    filters: HierarchicalFilter[]
+  ): string {
+    const filter = this.mapFilter(rawFilter, { ...initFilterProps } as HierarchicalFilter, context);
+    filter.children =
+      rawFilter.values?.map(
```

Comment:
If you have to `.map()` and `.push()` you can use the` .reduce()` in order to simplify the implementation.

Reply:
The `.map` is for the `rawFilter.values` in order to transform them into valid DTOs added to the `filter.children`.
The `.push` is for adding the filter which children have been mapped to the `filters` passed as reference.

The reduce version would be more complex.

```
    filter.children = rawFilter.values?.reduce((accFilters, rawFilterChild) => {
      accFilters.push(this.mapDeepFilter(rawFilterChild, { ...initFilterProps, parentId: filter.id }, context, filters)));
      return accFilters;
    }, []);
 ```
 
Or did you think something different?